src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -37,11 +37,329 @@
+    #try:
+    #    if MLContext.globalMLContext is None:
+    #        return __builtin__.print(*args, **kwargs)
+    #except AttributeError:
+    #    pass
+    if len(kwargs) > 0:
+        raise Exception('named argument not supported for mlprint inside context')
+    if len(args) > 0:
+        pydmlStmt = [getTabs(), 'print(', toString(args[0])]
+        for i in range(1, len(args)):
+            pydmlStmt = pydmlStmt + [' + ', toString(args[i])]
+        addPyDMLString(pydmlStmt + [')\n'])
+    else:
+            raise Exception('print cannot be empty')
+
+def toString(val):
+    if isinstance(val, Scalar):
+        return val.getID()
+    elif isinstance(val, Matrix):
+        return 'toString(' + val.getID() + ')'
+    else:
+        return '"' + str(val) + '"'
+        
+@contextmanager
+def parfor(s, start, end, check=None, par=None, mode=None, taskpartitioner=None, tasksize=None, datapartitioner=None, resultmerge=None, opt=None):
+    scopeCheck('parfor')
+    pydmlStmt = [getTabs(), 'parfor(', s.getID(), ' in ', getPythonFloatValOrScalarID(start), ':', getPythonFloatValOrScalarID(end)]
+    pydmlStmt = pydmlStmt + [getNamedArgOrEmpty('check', check), getNamedArgOrEmpty('par', par), getNamedArgOrEmpty('mode', mode), getNamedArgOrEmpty('taskpartitioner', taskpartitioner), getNamedArgOrEmpty('tasksize', tasksize), getNamedArgOrEmpty('datapartitioner', datapartitioner), getNamedArgOrEmpty('resultmerge', resultmerge), getNamedArgOrEmpty('opt', opt) ]
+    pydmlStmt = pydmlStmt + ['):\n']
+    addPyDMLString(pydmlStmt)
+    MLContext.globalMLContext.numTabs += 1
+    yield 
+    MLContext.globalMLContext.numTabs -= 1
+    pydmlStmt = [getTabs(), '\n']
+    addPyDMLString(pydmlStmt)
+
+@contextmanager
+def FOR(s, start, end):
+    scopeCheck('FOR')
+    pydmlStmt = [getTabs(), 'for(', s.getID(), ' in ', getPythonFloatValOrScalarID(start), ':', getPythonFloatValOrScalarID(end), '):\n']
+    addPyDMLString(pydmlStmt)
+    MLContext.globalMLContext.numTabs += 1
+    yield
+    MLContext.globalMLContext.numTabs -= 1
+    pydmlStmt = [getTabs(), '\n']
+    addPyDMLString(pydmlStmt)
+
+def getPythonFloatValOrScalarID(value):
+    if isinstance(value, float) or isinstance(value, int) or isinstance(value, long):
+        return str(float(value))
+    elif isinstance(value, Scalar):
+        return value.getID()
+    else:
+        raise Exception('Incorrect type: expected float or int or long or Scalar')
+        
+def scopeCheck(fnName):
+    if MLContext.globalMLContext is None:
+        raise Exception('Incorrect usage: \"' + fnName + '\" can only run if MLContext.start has been called')
+
+class random(object):
+    @staticmethod
+    def normal(rows=None, cols=None, sparsity=None):
+        if rows is None or cols is None:
+            raise Exception('rows and cols are mandatory arguments')
+        return simpleAddPyDMLStmt('random.normal', args=['rows=', str(rows), ', cols=', str(cols), ', sparsity=', getDefault(sparsity, 1.0)])
```

Comment:
`Remove the `getDefault` call, and just set `sparsity=1.0` in the function declaration for a default value for `sparsity`.